fuse/eval/metrics/metrics_common.py

```diff
@@ -887,6 +887,23 @@ def reset(self) -> None:
         self._metric.reset()
         return super().reset()
 
+    @staticmethod
+    def _convert_tuples(ids: List[Tuple[Any, ...]]) -> np.ndarray:
+
+        sample_tuple = ids[0]
+        dtype_tuple = []
+
+        for i, tuple_elem in enumerate(sample_tuple):
+            if isinstance(tuple_elem, str):
+                max_len = max(len(str(el[i])) for el in ids)
+                dtype_tuple.append((f"field{i}", f"U{max_len}"))
+            else:
+                dtype_tuple.append((f"field{i}", type(tuple_elem)))
+
+        ids = np.array(ids, dtype=dtype_tuple)
+        ids = [tuple(x) for x in ids]
+        return ids
```

Comment:
Are you returning here a list of tuples or numpy array?

Reply:
When applying only line 903 the results numpy array includes tuples but they have a problem of being hashable and I was getting error in the line :
permutation = [original_ids_pos[sample_id] for sample_id in required_ids] 
TypeError: unhashable type: 'writeable void-scalar'

The suggestion for the fix I found was to updated their types to tuple that is supported in the list and not in array. If I am trying to convert them after to np.array again the tuples are broken to separate elements and we having 2D array